### PR TITLE
quality_control: refactor confusion matrix initialization

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

The existing code creates a row and column for each category in the dataset, plus an extra row/column at the end for unmatched annotations. However, it's not immediately clear what order these rows/columns are in, because all access is done indirectly through the `confusion_matrix_labels_rmap` dictionary.

IMO, this is an unnecessary level of indirection, which obscures the structure of the matrix. It's easier to just assign rows/columns 0..(N-1) to the categories and N to "unmatched".

Do just that. Additionally, factor out the code creating the matrix into a function, because it's used in two places.

### How has this been tested?

Manual testing.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/quality_control/quality_reports.py [hunk 1]
```diff
@@ -2025,15 +2012,11 @@ def generate_report(self) -> ComparisonReport:
         label_recalls = _arr_div(matched_ann_counts, gt_ann_counts)
 
         valid_annotations_count = np.sum(matched_ann_counts)
-        missing_annotations_count = np.sum(confusion_matrix[confusion_matrix_labels_rmap[None], :])
-        extra_annotations_count = np.sum(confusion_matrix[:, confusion_matrix_labels_rmap[None]])
+        missing_annotations_count = np.sum(confusion_matrix[-1, :])
```
Comment (zhiltsov-max) [inline cvat/apps/quality_control/quality_reports.py:2015]: Why the matrix accumulation code was removed?

File: cvat/apps/quality_control/quality_reports.py [hunk 2]
```diff
@@ -1906,15 +1896,11 @@ def _find_closest_unmatched_shape(shape: dm.Annotation):
...
```
Comment (zhiltsov-max) [inline cvat/apps/quality_control/quality_reports.py:1899]: It seems like now this -1 is a good candidate for a named constant, as it's used in several places.

File: cvat/apps/quality_control/quality_reports.py [hunk 3]
```diff
@@ -1994,27 +1994,14 @@ def generate_report(self) -> ComparisonReport:
...
 
-        confusion_matrix_labels = {
-            i: label.name
-            for i, label in enumerate(self._gt_dataset.categories()[dm.AnnotationType.label])
-            if not label.parent
-        }
-        confusion_matrix_labels[None] = "unmatched"
-        confusion_matrix_labels_rmap = {k: i for i, k in enumerate(confusion_matrix_labels.keys())}
-        if confusion_matrices:
-            confusion_matrix = np.sum(confusion_matrices, axis=0)
```
Comment (zhiltsov-max) [inline cvat/apps/quality_control/quality_reports.py:2013]: Why the matrix accumulation code was removed?

Comment (SpecLad) [reply cvat/apps/quality_control/quality_reports.py:2013]: It's not removed, it's just done in the loop above now.

Comment (zhiltsov-max) [reply cvat/apps/quality_control/quality_reports.py:2013]: It's less efficient in the new variant, I suppose. There can be quite a few frames in the task, and the old variant summed them with numpy.

File: cvat/apps/quality_control/quality_reports.py (hunk 2 above)
Comment (SpecLad) [reply cvat/apps/quality_control/quality_reports.py]: Yeah, fair point. Replaced with a constant.

File: cvat/apps/quality_control/quality_reports.py (hunk 3 above)
Comment (SpecLad) [reply cvat/apps/quality_control/quality_reports.py:2013]: Okay, I did a little benchmark, and at matrix sizes where the time taken non-negligible the new variant is faster. 🤷

Comment (zhiltsov-max) [reply cvat/apps/quality_control/quality_reports.py:2013]: Looks strange, but if so, okay. Maybe it's better to just include this in the `accumulate()` then. There is also a parallel [PR](https://github.com/opencv/cvat/pull/7604) that touches this topic.

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**